server/service/osquery.go

```diff
@@ -1409,7 +1451,33 @@ func (svc *Service) SubmitResultLogs(ctx context.Context, logs []json.RawMessage
 	// skipauth: Authorization is currently for user endpoints only.
 	svc.authz.SkipAuthorization(ctx)
 
-	if err := svc.osqueryLogWriter.Result.Write(ctx, logs); err != nil {
+	//
+	// We do not return errors to osqueryd when processing results because
+	// otherwise the results will never clear from its local DB and
+	// will keep retrying forever.
+	//
+
+	unmarshaledResults, queriesDBData := svc.preProcessOsqueryResults(ctx, logs)
+	svc.saveResultLogsToQueryReports(ctx, unmarshaledResults, queriesDBData)
+
+	var filteredLogs []json.RawMessage
+	for i, unmarshaledResult := range unmarshaledResults {
+		if unmarshaledResult == nil {
+			// Ignore results that could not be unmarshaled.
+			continue
+		}
+		dbQuery, ok := queriesDBData[unmarshaledResult.QueryName]
+		if !ok {
+			// Ignore results for unknown queries.
+			continue
```

Comment:
sensible, but a change in behavior from today

Reply:
Indeed. I asked this question to @zwass on Slack.

Reply:
From Zach:
> I don't anticipate that causing any issues but I recommend we document it in the release notes. Perhaps something like "Fleet now does more validation on osquery logs that are ingested and will error on unknown message formats."